src/js/components/DateInput/DateInput.js

```diff
@@ -275,14 +278,20 @@ Use the icon prop instead.`,
                   reference,
                   outputFormat,
                 );
-                if (nextValue !== undefined)
-                  setReference(getReference(nextValue));
+
+                const validatedNextValue = validateDateBounds({
```

Comment:
```suggestion
                const validatedNextValue = validateDateBounds(
```
Can we just pass these as function parameters instead of an object?

Reply:
done